### DISCUSSION TITLE
Upload to custom provider

### DISCUSSION THREAD

**Discussion by SantoshSrinivas79** [uploadcare/file-uploader#561] [blocks]
Is there a way to use the client library and upload to S3 or Cloudflare etc?

**Comment by nd0ut** [comment]
Hey @SantoshSrinivas79,


Uploader is only compatible with the Uploadcare Upload API.

Could you clarify? Are you looking to upload files to a different storage while still using Uploadcare CDN features, or do you want to switch both storage and delivery?

**Reply by SantoshSrinivas79** [reply]
Hi @nd0ut .... I am running out of space for my side projects. Was wondering if I could move to some other service while continuing to use the uploader. I'd love to keep everything in Uploadcare if the projects get some revenue
